pkg/entity/register/worker.go

```diff
@@ -60,47 +73,99 @@ func (w *worker) Run(ctx context.Context) {
 
 			if batchSize == w.maxBatchSize {
 				timer.Reset(w.maxBatchDuration)
-				w.send(batch, &batchSize)
+				w.send(ctx, batch, &batchSize)
 			}
 
 		case <-timer.C:
 			if len(batch) > 0 {
-				w.send(batch, &batchSize)
+				w.send(ctx, batch, &batchSize)
 			}
 			timer.Reset(w.maxBatchDuration)
 		}
 	}
 }
 
-func (w *worker) send(batch map[entity.Key]fwrequest.EntityFwRequest, batchSize *int) {
+func (w *worker) send(ctx context.Context, batch map[entity.Key]fwrequest.EntityFwRequest, batchSize *int) {
 	defer w.resetBatch(batch, batchSize)
 
 	var entities []entity.Fields
 	for _, r := range batch {
 		entities = append(entities, r.Data.Entity)
 	}
-	responses, _, errClient := w.client.RegisterBatchEntities(w.agentIDProvide().ID, entities)
-	if errClient != nil {
-		// TODO error handling
-		return
-	}
+
+	responses := w.registerEntitiesWithRetry(ctx, entities)
```

Comment:
I'm wondering if this will make the reg synchronous, since we are waiting for the response.

Not sure if this is expected.

Reply:
nope, this is within "worker" scope, therefore only such worker gets blocked